tc/sgx/trusted_worker_manager/enclave/kme/CMakeLists.txt

```diff
@@ -0,0 +1,76 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+CMAKE_MINIMUM_REQUIRED(VERSION 3.2 FATAL_ERROR)
+PROJECT(libavalon-kme-enclave  CXX)
+
+INCLUDE(../CMakeVariables.txt)
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+# ADDITIONAL LIBRARIES
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+SET(TCF_TOP_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../../../../")
+SET(TCF_CORE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../../")
+#SET(WPE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/wpe)
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+FILE(GLOB PROJECT_HEADERS *.h ../*.h)
+FILE(GLOB PROJECT_SOURCES *.cpp *.c ../*.cpp)
+FILE(GLOB PROJECT_EDL kme_enclave.edl)
+FILE(GLOB PROJECT_CONFIG *.xml)
+SOURCE_GROUP("Source" FILES ${PROJECT_HEADERS} ${PROJECT_SOURCES} ${PROJECT_EDL})
+
+SET(EDL_PATH ${CMAKE_CURRENT_SOURCE_DIR})
+
+SGX_EDGE_TRUSTED(${PROJECT_EDL} ${EDL_PATH} PROJECT_EDGE_SOURCES)
+
+SOURCE_GROUP("Generated" FILES ${PROJECT_EDGE_SOURCES})
+
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+# Compile targets
+# XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+# CMake likes to prepend "lib" to project name when creating library...since we
+# already have lib in the project name, liblibavalon-singleton-enclave is stupid.  Stop
+# CMake from doing it.
+SET(CMAKE_SHARED_LIBRARY_PREFIX "")
+PROJECT(libavalon-kme-enclave C CXX)
+
+ADD_LIBRARY(${PROJECT_NAME} SHARED ${PROJECT_HEADERS} ${PROJECT_SOURCES} ${PROJECT_EDGE_SOURCES} ${PROJECT_EDL})
+SGX_SIGN_ENCLAVE(${PROJECT_NAME} ${TCF_ENCLAVE_CODE_SIGN_PEM} ${PROJECT_CONFIG})
+
+# Configure CPP linker flags
+SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/avalon_kme_enclave.lds")
+# Include KME workload library path
+SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-L,${CMAKE_CURRENT_SOURCE_DIR}/workload/build")
+
+TARGET_LINK_LIBRARIES(${PROJECT_NAME} -Wl,--whole-archive -lsgx_tsgxssl -Wl,--no-whole-archive)
+TARGET_LINK_LIBRARIES(${PROJECT_NAME} -Wl,--whole-archive -l${TRTS_LIBRARY_NAME} -Wl,--no-whole-archive)
+
+TARGET_LINK_LIBRARIES(${PROJECT_NAME} -Wl,--whole-archive -lcommon_sgx_workload
+-lcommon_sgx_iohandler -Wl,--no-whole-archive)
+
+TARGET_LINK_LIBRARIES(${PROJECT_NAME} -Wl,--start-group  -ltavalon-common
+-ltavalon-base64 -ltavalon-parson -ltavalon-crypto -ltavalon-verify-ias-report
+-ltavalon-sgx-common -lsgx_tsgxssl_crypto -lsgx_tstdc -lsgx_tcxx -lsgx_tcrypto
+-l${SERVICE_LIBRARY_NAME} -Wl,--end-group)
```

Comment:
Above are common and an be moved to CMakeVariables.txt

Reply:
Created a function LINK_COMMON_LIBRARIES() in CMakeVariables.txt and moved these statements to function.